[shelly] Use thing type `shellyplusht` for Shelly Plus HT Gen3

markus7017 opened the pull request:
There is no need to have an extra thing shellyplushtg3. This is just a new hardware revision of the HT. As with other devices the different hw revisions get mapped to the base thing avoiding creating a new thing type for every hw revision/generation.

BREAKING CHANGE: thing type shellyplushtg3 is removed, instead a thing of type shellyplusg3 gets created

@wborm fyi

jlaur commented:
The thing type being removed here was introduced in 4.2 by #16625.

lsiepel commented:
More to remove if this is really needed: https://github.com/openhab/openhab-addons/pull/18939#discussion_r2203493704

jlaur commented:
Starting point: https://github.com/openhab/openhab-addons/pull/18925#issuecomment-3067124195

@markus7017 - did you mean it was removed again by a prior PR and then reintroduced? This would be good to know.

jlaur commented on bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/shellyGen2_sensor.xml:
```diff
@@ -19,21 +19,6 @@
 		<config-description-ref uri="thing-type:shelly:battery-gen2"/>
 	</thing-type>
 
-	<thing-type id="shellyhtg3">
```
The reference in the README should also be removed:
```
### Shelly Plus HT (thing-type: shellyplusht), Plus HT Gen 3 (thing-type: shellyhtg3)
```

jlaur commented:
Just for my understanding: The reason we don't need this thing type is because we have another thing type using the same API version - is that correct? That's what I got from https://github.com/openhab/openhab-addons/pull/18775#issuecomment-2961204992. So if we map model `S3SN-0U12A` to thing type `shellyplusht`, then we are fine, because it already uses API v2?

Next, since we already had it mapped this way:

https://github.com/openhab/openhab-addons/blob/42527780f1d44f7d7be90a1bace3c2407ee51bfb/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyThingCreator.java#L310-L311

it shouldn't be a breaking change, just a cleanup, since the thing type was not referenced. And now I finally realize that you already changed the mapping in #18782 so that `S3SN-0U12A` now maps to `shellyplusht` rather than  `shellyplushtg3`. So the breaking change is already merged, and this is the cleanup of the no longer used pieces.

Correct me if my analysis is wrong. 😄

jlaur reviewed (changes requested):
Besides the README, it looks like you might have missed to stage `ShellyDeviceProfile` and `ShellyDeviceProfileTests` which still references the deleted `THING_TYPE_SHELLYPLUSHTG3`, and hence the build fails. In the latter, please replace:

```java
                Arguments.of(THING_TYPE_SHELLYPLUSHTG3, true, false), //
```

by:
```java
                Arguments.of(THING_TYPE_SHELLYPLUSHT, true, false), //
```

since this coverage was missing.

markus7017 commented:
@jlaur The shellyplushtg3 thing came in here: #18814. I don't know that this was part of the last milestone build so if it's a breaking change or not. Nevertheless, why not point out for the user, even we have already an update alert and I added info on plushtg3 too (PR #1767) and set @wborn  fyi.

Yes, g3 has the V2 API and we habe the DT to shellyplusht as we do for other devices, e.g. Plus2PM g4 now also maps to shellyplus2pm.

I fixed the build problems. I can't find a remaining reference in README. Did you checked main branch, or this PR?

jlaur commented on bundles/org.openhab.binding.shelly/src/test/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfileTest.java:
```diff
@@ -119,7 +119,6 @@ private static Stream<Arguments> provideTestCasesForApiDetermination() {
                 Arguments.of(THING_TYPE_SHELLYPRO4PM, true, false), //
                 Arguments.of(THING_TYPE_SHELLYPLUSDIMMERUS, true, false), //
                 Arguments.of(THING_TYPE_SHELLYPLUSDIMMER10V, true, false), //
-                Arguments.of(THING_TYPE_SHELLYPLUSHTG3, true, false), //
```
See https://github.com/openhab/openhab-addons/pull/18943#pullrequestreview-3014350077

Please keep this test for `THING_TYPE_SHELLYPLUSHT`.

markus7017 commented:
> > I can't find a remaining reference in README. Did you checked main branch, or this PR?
> 
> Just search for "shellyhtg3" - you'll find it here (it's in the main branch): https://next.openhab.org/addons/bindings/shelly/#shelly-plus-ht-thing-type-shellyplusht-plus-ht-gen-3-thing-type-shellyhtg3

I can't find it here (main branch), looks good for me
https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.shelly

jlaur commented:
> > Just search for "shellyhtg3" - you'll find it here (it's in the main branch): https://next.openhab.org/addons/bindings/shelly/#shelly-plus-ht-thing-type-shellyplusht-plus-ht-gen-3-thing-type-shellyhtg3
> 
> I can't find it here (main branch), looks good for me https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.shelly

I don't have anything to add. I don't know why you are not able to find it on this exact link you sent. 🤷‍♂️

jlaur commented:
Two open comments to resolve and one merge conflict. I'll be unavailable after tomorrow, and the code freeze for RC1 is on Friday.

lsiepel commented:
Sorry @markus7017 it didn;t make it in. The RC1 build just started and this PR is not ready for a merge as there are build issues.

markus7017 commented:
@jlaur build bug is fixed, did some testing
ready to merge

jlaur reviewed (changes requested):
The conflicts were not resolved correctly.
